modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQConsumer.java

```diff
@@ -34,255 +34,220 @@
 import java.io.ObjectInput;
 import java.io.ObjectInputStream;
 
+/**
+ * The message consumer responsible for read a message from the queue and set it into the message context
+ */
 public class RabbitMQConsumer implements MessageConsumer {
-	private static final Log logger = LogFactory.getLog(RabbitMQConsumer.class.getName());
-
-	private Connection connection;
-
-	private Channel channel;
-
-	private RabbitMQStore store;
-
-	private String queueName;
-
-	private String idString;
-
-	private boolean isInitialized;
-	/**
-	 * Did last receive() call cause an error?
-	 */
-	private boolean isReceiveError;
-	/**
-	 * Holds the last message read from the message store.
-	 */
-	private CachedMessage cachedMessage;
-
-	public RabbitMQConsumer(RabbitMQStore store) {
-		if (store == null) {
-			logger.error("Cannot initialize.");
-			return;
-		}
-		this.store = store;
-		cachedMessage = new CachedMessage();
-		isReceiveError = false;
-		isInitialized = true;
-	}
-
-	public MessageContext receive() {
-		if (!checkConnection()) {
-			if (!reconnect()) {
-				if (logger.isDebugEnabled()) {
-					logger.debug(getId()
-					             + " cannot receive message from store. Can not reconnect.");
-				}
-				return null;
-			} else {
-				logger.info(getId() + " reconnected to store.");
-				isReceiveError = false;
-			}
-		}
-		//setting channel
-		if (channel != null) {
-			if (!channel.isOpen()) {
-				if (!setChannel()) {
-					logger.info(getId() + " unable to create the channel.");
-					return null;
-				}
-			}
-		} else {
-			if (!setChannel()) {
-				logger.info(getId() + " unable to create the channel.");
-				return null;
-			}
-		}
-		//receive messages
-		try {
 
-			GetResponse delivery = null;
-			delivery = channel.basicGet(queueName, false);
-
-			if (delivery != null) {
-				//deserilizing message
-				StorableMessage storableMessage = null;
-				ByteArrayInputStream bis = new ByteArrayInputStream(delivery.getBody());
-				ObjectInput in = new ObjectInputStream(bis);
-				try {
-					storableMessage = (StorableMessage) in.readObject();
-				} catch (ClassNotFoundException e) {
-					logger.error(getId() + "unable to read the stored message" + e);
-					channel.basicAck(delivery.getEnvelope().getDeliveryTag(), false);
-				}
+    private static final Log log = LogFactory.getLog(RabbitMQConsumer.class.getName());
+
+    private Connection connection;
+    private Channel channel;
+    private RabbitMQStore store;
+    private String queueName;
+    private String idString;
+    private CachedMessage cachedMessage; // Holds the last message read from the message store
+
+    public RabbitMQConsumer(RabbitMQStore store) {
+        if (store == null) {
+            log.error("Cannot initialize.");
```

Comment:
Cannot initialise what? Shall we say 
`Cannot initialise RabbitMQ message processor + message processor name`? 
My suggestion is to handle a situation using exceptions rather returning null. During deployment, this should fail the deployment (ultimately the CAR file deployment fails)

Reply:
Ack to improve the error log message. The `return;` statement used not to initialize the constructor upon the store becomes null. That is how it was handled in other types of store processor as well. Technically, the store won't become null if you deploy the processor by creating through management console or tooling.

https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsConsumer.java#L66

Reply:
The store can never be null right? It's called inside RabbitMQStore.getConsumer() passing (this). In that case, we can remove the if block?

Reply:
Yes. It was there in other implementation as well. There could be a situation where somebody is copying message processor synapse config to the deployment without messages store synapse config. I guess it won't make any harm just keeping that as it is.